lib/modules/manager/pep621/processors/pdm.ts

```diff
@@ -0,0 +1,42 @@
+import is from '@sindresorhus/is';
+import { PypiDatasource } from '../../../datasource/pypi';
+import type { PackageDependency } from '../../types';
+import type { PyProject } from '../schema';
+import { parseDependencyGroupRecord } from '../utils';
+import type { PyProjectProcessor } from './types';
+
+export class PdmProcessor implements PyProjectProcessor {
+  process(project: PyProject, deps: PackageDependency[]): PackageDependency[] {
+    const pdm = project.tool?.pdm;
+    if (is.nullOrUndefined(pdm)) {
+      return deps;
+    }
+
+    deps.push(
+      ...parseDependencyGroupRecord(
+        'tool.pdm.dev-dependencies',
+        pdm['dev-dependencies']
+      )
+    );
+
+    const pdmSource = pdm.source;
+    if (is.nullOrUndefined(pdmSource)) {
+      return deps;
+    }
+
+    // add pypi default url, if there is no source declared with the name `pypi`. https://daobook.github.io/pdm/pyproject/tool-pdm/#specify-other-sources-for-finding-packages
+    const containsPyPiUrl = pdmSource.some((value) => value.name === 'pypi');
+    const registryUrls: string[] = [];
+    if (!containsPyPiUrl) {
+      registryUrls.push(PypiDatasource.defaultURL);
+    }
+    for (const source of pdmSource) {
+      registryUrls.push(source.url);
+    }
+    for (const dep of deps) {
+      dep.registryUrls = registryUrls;
```

Comment:
shouldn't we set it on results object once for all reps?

Reply:
This additional source urls are specific to PDM 
https://pdm.fming.dev/latest/usage/config/#configure-the-package-indexes

Reply:
sure, but you override the registry for all deps anyways, so i thought to pass the result object to the processor, so it can set the registryUrls on the result instead of changing every dep.

Or do you think some deps can have different urls later? 🤔

Reply:
That I override this for all dependencies is intended, as these source urls should also affect the standard deps e.g. from `project.dependencies`.